src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java

```diff
@@ -140,16 +137,13 @@ public void setWorkingDir(String workingDir) {
         this.workingDir = workingDir;
     }
 
-    @Extension
-    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
-
-        public DescriptorImpl() {
-            super(PodTemplateStepExecution.class);
-        }
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new PodTemplateStepExecution(this, context);
```

Comment:
I guess this answers my previous question: `why do we prefer to get FilePath from context instead of having it injected?`.

Still its not clear to me why do we do that?